doc/source/ray-core/miscellaneous.rst

```diff
@@ -216,3 +216,135 @@ To get information about the current available resource capacity of your cluster
 
 .. autofunction:: ray.available_resources
     :noindex:
+
+Running Large Ray Clusters
+--------------------------
+
+Here are some tips to run Ray with more than 1k nodes. When running Ray with such
+a large number of nodes, several system settings may need to be tuned to enable
+communication between such a large number of machines.
+
+Tuning Operating System Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Because all nodes and workers connect to the GCS, many network connections will
+be created and the operating system has to support that number of connections.
+
+Maximum open files
+******************
+
+The OS has to be configured to support opening many TCP connections since every
+worker and raylet connects to the GCS. In POSIX systems, the current limit can
+be checked by `ulimit -n` and if it's small, it should be increased according to
+the OS manual. 
+
+ARP cache
+*********
+
+Another thing that needs to be configured is the ARP cache. In a large cluster,
+all the worker nodes connect to the head node, which adds a lot of entries to
+the ARP table. Ensure that the ARP cache size is large enough to handle this
+many nodes.
+Failure to do this will result in the head node hanging. When this happens,
+`dmesg` will show errors like `neighbor table overflow message`.
+
+In Ubuntu, the ARP cache size can be tuned in `/etc/sysctl.conf` by increasing
+the value of `net.ipv4.neigh.default.gc_thresh1` - `net.ipv4.neigh.default.gc_thresh3`. 
+For more details, please refer to the OS manual.
+
+Tuning Ray Settings
+~~~~~~~~~~~~~~~~~~~
+
+To run a large cluster, several parameters need to be tuned in Ray.
+
+Health check
+************
+
+GCS checks the health of the worker nodes by sending ping periodically.
+Sometimes, if the network is poor or the raylet is too busy to respond,
+the raylet might be marked dead incorrectly. 
+
+GCS uses a similar algorithm as k8s probes to check the healthiness of the
+Raylet: it sends the ping to the worker periodically and if it failed X times
+consecutively, the worker will be marked as dead. Once a worker is marked as
+dead, it'll be considered dead forever even it's not. The following OS
+environments can be used to tune this: 
+
+- `RAY_health_check_initial_delay_ms` The delay of the first health check, 5000ms by default.
+- `RAY_health_check_period_ms` The interval between two health check requests, 3000ms by default.
+- `RAY_health_check_timeout_ms` The timeout to consider a health check failed, 10000ms by default.
+- `RAY_health_check_failure_threshold` The consecutive failure threshold for the
```

Comment:
Since we switched to pull-based health checking, do we need to mention this? It seems misleading if it isn't an issue in practice now.

Reply:
I think as long as raylet is not overloaded, it's ok. (open to remote this for now)